osg/short-term.md

```diff
@@ -0,0 +1,292 @@
+```
+
+## Job Config Files
+
+The job configuration file is a JSON file named `config.json` containing configuration settings for the job. The
+following configuration settings are provided:
+
+### arguments
+
+This setting is an array of strings containing the command-line arguments generated by the DE based on the app
+configuration and the parameter values selected by the user. In cases where the wrapper script runs the tool itself as a
+subprocess, this argument list can generally be passed to the subprocess as-is. In other cases, the wrapper script must
+interpret the argument array directly.
+
+### irods_host
+
+This setting contains the host name to use when connecting to iRODS.
+
+### irods_port
+
+This setting contains the port number to use when connecting to iRODS.
+
+### irods_job_user
+
+This setting contains the username of the person who submitted the job. This user should have ownership permissions on
+the output files after the job completes.
+
+### irods_user
+
+This setting contains the username to use when authenticating to iRODS. This setting can be ignored if you're using
+custom iRODS credentials.
+
+### input_ticket_list
+
+This setting contains the name of the input ticket list file.
+
+### output_ticket_list
+
+This setting contains the name of the output ticket list file.
+
+### status_update_url
+
+This setting contains a URL that should be used to send job status updates back to the DE. This will be described in
+more detail later.
+
+### stdout
+
+This setting contains the name of the file that should contain the redirected standard output produced by the tool.
+
+### stderr
+
+This setting contains the name of the file that should contain the redirected standard error output produced by the
+tool.
+
+## Initializing the iRODS Connection
+
+Initializing the iRODS connection is actually fairly simple if you're using version 4.x of the icommands, which is
+highly recommended. The username in the `irods_user` configuration setting is guaranteed to have an empty password, so
+iRODS can be initialized by creating a file called `$HOME/.irods/irods_environment.json`. This file should be in the
+following format:
+
+```
+{
+    "irods_user_name": "<username>",
+    "irods_host": "<host-name>",
+    "irods_port": <port-number>,
+    "irods_zone_name": ""
+}
+```
+
+For the time being, the zone name should always be equal to the empty string. The rest of the settings are extracted
+from the job configuration file. After creating this file, `iget` and `iput` can be used in conjunction with the iRODS
+tickets to retrieve files from or push files to iRODS.
+
+If you're using custom iRODS credentials with a password then (assuming you're not using Jargon) `iinit` must be called
+before using any of the iRODS icommands. `iinit` only runs in interactive mode, so it will be necessary to send the
+password to `iinit` using redirection. Using custom iRODS credentials is expected to be rare, so precise instructions
```

Comment:
If we create a non-anonymous (but still basically "anonymous") user and pass that in the config.json (understanding, of course, that we may not get to that before we push this out), I think the implementor would still need to run `iinit` unless I'm mistaken, so we may want to add more precise instructions if that happens. (This may already be in your mind but I feel like I should be complete when doing reviews, heh)

Reply:
Running the commands without running `iinit` for our non-anonymous account actually works in our `osg-word-count` image. It does spew a bunch of warning messages, though. 😁